How to enable Git authentication through credentials passed as environment GRGIT_USER and GRGIT_PASS

Using the latest release-opinion plugin
```
plugins {
    id 'org.ajoberstar.grgit' version '1.7.0'
    id 'org.ajoberstar.release-opinion' version '1.7.0'
```


A Jenkins pipeline build sets the GRGIT_USER and GRGIT_PASS variables and invokes the release task.

`./gradlew  -Prelease.scope=patch -Prelease.stage=milestone -Dorg.ajoberstar.grgit.auth.force=hardcoded  --debug --stacktrace release
`

However the authentication code takes the decision to try using system properties and fails.

```
18:01:17.532 [INFO] [org.ajoberstar.grgit.auth.TransportOpUtil] The following authentication options are allowed (though they may not be available): [HARDCODED]
18:01:17.540 [INFO] [org.ajoberstar.grgit.auth.TransportOpUtil] using hardcoded credentials from system properties
18:01:17.547 [DEBUG] [org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter] Removed task artifact state for {} from context.
18:01:17.547 [DEBUG] [org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter] Finished executing task ':prepare'
18:01:17.547 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] :prepare FAILED
18:01:17.548 [INFO] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] :prepare (Thread[main,5,main]) completed. Took 0.067 secs.
18:01:17.551 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationWorkerRegistry] Worker root.1 completed (0 in use)
18:01:17.551 [DEBUG] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] Task worker [Thread[main,5,main]] finished, busy: 0.067 secs, idle: 0.004 secs
18:01:17.562 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
18:01:17.565 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
18:01:17.566 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
18:01:17.567 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
18:01:17.568 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':prepare'.
18:01:17.572 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Ambiguous method overloading for method org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider#<init>.
18:01:17.573 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   Cannot resolve which method to invoke for [class java.lang.String, null] due to overlapping prototypes between:
18:01:17.574 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   	[class java.lang.String, class java.lang.String]
18:01:17.574 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   	[class java.lang.String, class [C]

```

Only by adding `-Dorg.ajoberstar.grgit.auth.username=$GRGIT_USER -Dorg.ajoberstar.grgit.auth.password=$GRGIT_PASS` to the gradle invocation Git authentication is working.

According to http://ajoberstar.org/grgit/docs/groovydoc/org/ajoberstar/grgit/auth/AuthConfig.html environment variables should count toward hardcoded authentication?

--- a/build.gradle
+++ b/build.gradle
@@ -42,7 +42,7 @@ dependencies {
     compile gradleApi()
 
     // grgit
-    compile 'org.ajoberstar:grgit:1.9.0'
+    compile 'org.ajoberstar:grgit:1.9.1'
 
     // semver
     compile 'com.github.zafarkhaja:java-semver:0.9.0'

--- a/src/main/groovy/org/ajoberstar/gradle/git/release/semver/RebuildVersionStrategy.groovy
+++ b/src/main/groovy/org/ajoberstar/gradle/git/release/semver/RebuildVersionStrategy.groovy
@@ -53,14 +53,14 @@ class RebuildVersionStrategy implements VersionStrategy {
     @Override
     boolean selector(Project project, Grgit grgit) {
         def clean = grgit.status().clean
-        def props = project.properties.keySet().find { it.startsWith('release.') }
+        def propsSpecified = project.hasProperty(SemVerStrategy.SCOPE_PROP) || project.hasProperty(SemVerStrategy.STAGE_PROP)
         def headVersion = getHeadVersion(project, grgit)
 
-        if (clean && props == null && headVersion) {
+        if (clean && !propsSpecified && headVersion) {
             logger.info('Using {} strategy because repo is clean, no "release." properties found and head version is {}', name, headVersion)
             return true
         } else {
-            logger.info('Skipping {} strategy because clean is {}, "release." properties are {} and head version is {}', name, clean, props, headVersion)
+            logger.info('Skipping {} strategy because clean is {}, "release." properties are {} and head version is {}', name, clean, propsSpecified, headVersion)
             return false
         }
     }

--- a/src/test/groovy/org/ajoberstar/gradle/git/release/semver/RebuildVersionStrategySpec.groovy
+++ b/src/test/groovy/org/ajoberstar/gradle/git/release/semver/RebuildVersionStrategySpec.groovy
@@ -51,7 +51,7 @@ class RebuildVersionStrategySpec extends Specification {
     def 'selector returns false if any release properties are set'() {
         given:
         mockClean(true)
-        Project project = getProject('release.anything': 'value')
+        Project project = getProject('release.scope': 'value')
         mockTagsAtHead('v1.0.0')
         expect:
         !strategy.selector(project, grgit)

